src/CompletionProvider.php

```diff
@@ -0,0 +1,383 @@
+                    // Find the shortest name to reference the symbol
+                    if ($namespace && ($alias = array_search($def, $aliasedDefs, true)) !== false) {
+                        // $alias is the name under which this definition is aliased in the current namespace
+                        $item->insertText = $alias;
+                    } else if ($namespace && !($prefix && $isFullyQualified)) {
+                        // Insert the global FQN with trailing backslash
+                        $item->insertText = '\\' . $fqn;
+                    } else {
+                        // Insert the FQN without trailing backlash
+                        $item->insertText = $fqn;
+                    }
+                    $items[] = $item;
+                }
+            }
+            // Suggest keywords
+            if ($node instanceof Node\Name && $node->getAttribute('parentNode') instanceof Node\Expr\ConstFetch) {
+                foreach (self::KEYWORDS as $keyword) {
+                    if (substr($keyword, 0, $prefixLen) === $prefix) {
+                        $item = new CompletionItem($keyword, CompletionItemKind::KEYWORD);
+                        $item->insertText = $keyword . ' ';
+                        $items[] = $item;
+                    }
+                }
+            }
+        } else if (
+            $node instanceof Node\Expr\Variable
+            || ($node && $node->getAttribute('parentNode') instanceof Node\Expr\Variable)
+         ) {
+            // Find variables, parameters and use statements in the scope
+            // If there was only a $ typed, $node will be instanceof Node\Error
+            $namePrefix = $node instanceof Node\Expr\Variable && is_string($node->name) ? $node->name : '';
+            foreach ($this->suggestVariablesAtNode($node, $namePrefix) as $var) {
+                $item = new CompletionItem;
+                $item->kind = CompletionItemKind::VARIABLE;
+                $item->label = '$' . ($var instanceof Node\Expr\ClosureUse ? $var->var : $var->name);
+                $item->documentation = $this->definitionResolver->getDocumentationFromNode($var);
+                $item->detail = (string)$this->definitionResolver->getTypeFromNode($var);
+                $item->textEdit = new TextEdit(
+                    new Range($pos, $pos),
+                    stripStringOverlap($doc->getRange(new Range(new Position(0, 0), $pos)), $item->label)
+                );
+                $items[] = $item;
+            }
+        } else if ($node instanceof Node\Stmt\InlineHTML || $pos == new Position(0, 0)) {
+            $item = new CompletionItem('<?php', CompletionItemKind::KEYWORD);
+            $item->textEdit = new TextEdit(
+                new Range($pos, $pos),
+                stripStringOverlap($doc->getRange(new Range(new Position(0, 0), $pos)), '<?php')
+            );
+            $items[] = $item;
+        }
+
+        return $items;
```

Comment:
Please return `CompletionList` instead of `CompletionItem[]`. Due to https://github.com/Microsoft/language-server-protocol/issues/39 the Java binding supports only `CompletionList`.